pkg/datapath/linux/datapath.go

```diff
@@ -38,19 +39,21 @@ type linuxDatapath struct {
 	nodeAddressing datapath.NodeAddressing
 	config         DatapathConfiguration
 	loader         *loader.Loader
+	wgAgent        *wireguard.Agent
 }
 
 // NewDatapath creates a new Linux datapath
-func NewDatapath(cfg DatapathConfiguration, ruleManager datapath.IptablesManager) datapath.Datapath {
+func NewDatapath(cfg DatapathConfiguration, ruleManager datapath.IptablesManager, wgAgent *wireguard.Agent) datapath.Datapath {
```

Comment:
Shouldn't the datapath define a local interface to abstract away the real `wireguard.Agent`? This should reduce the imports of the package where it's unnecessary

Reply:
Good feedback!